dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -218,4 +240,295 @@ public Response deleteEnvironmentPushHistory(@Context HttpServletRequest request
+            }
+        });
+
+        return Response.ok(new ResponseEntityView(
+                "Removing bundles in a separated process, the result of the operation will be notified")).build();
+    } // deleteBundlesByIdentifiers.
+
+    private void sendErrorDeleteBundleMessage(final InitDataObject initData,
+                                              final Locale locale,
+                                              final Exception e) {
+
+        final String message = Try.of(()->LanguageUtil.get(locale, "bundle.deleted.error.msg", e.getMessage()))
+                .onFailure(ex -> Logger.error(this, e.getMessage()))
+                .getOrElse("An error occurred deleting bundles, please check the log, error message: " + e.getMessage());
+
+        this.systemMessageEventUtil.pushMessage(new SystemMessageBuilder()
+                .setMessage(message)
+                .setLife(DateUtil.TEN_SECOND_MILLIS)
+                .setSeverity(MessageSeverity.ERROR).create(), Collections.singletonList(initData.getUser().getUserId()));
+    }
+
+    private void sendSuccessDeleteBundleMessage(final int bundleDeletesSize,
+                                                final InitDataObject initData,
+                                                final Locale locale) throws DotDataException {
+
+	    final String userId  = initData.getUser().getUserId();
+        final String message = Try.of(()->LanguageUtil.get(locale, "bundle.deleted.success.msg", bundleDeletesSize))
+                .onFailure(e -> Logger.error(this, e.getMessage()))
+                .getOrElse(bundleDeletesSize + " Bundles Deleted Successfully");
+
+        this.systemMessageEventUtil.pushMessage(new SystemMessageBuilder()
+                .setMessage(message)
+                .setLife(DateUtil.SEVEN_SECOND_MILLIS)
+                .setSeverity(MessageSeverity.INFO).create(), Collections.singletonList(userId));
+
+        APILocator.getSystemEventsAPI().push(SystemEventType.DELETE_BUNDLE,
+                new Payload(
+                        message,
+                        Visibility.USER,
+                        userId
+                )
+        );
+    }
+
+    // one transaction for each bundle
+    private void deleteBundleByIdentifier(final DeleteBundlesByIdentifierForm deleteBundlesByIdentifierForm, final InitDataObject initData) throws DotDataException {
+
+        for (final String bundleId : deleteBundlesByIdentifierForm.getIdentifiers()) {
+
+            this.bundleAPI.deleteBundleAndDependencies(bundleId, initData.getUser());
+        }
+    }
+
+    /**
+     * Deletes bundles older than a date. (unsent are not going to be deleted)
+     * Note: the response will be notified by socket message
+     * @param request   {@link HttpServletRequest}
+     * @param response  {@link HttpServletResponse}
+     * @param olderThan {@link ISODateParam} an ISO date, should be before now to be valid
+     */
+    @DELETE
+    @Path("/olderthan/{olderThan}")
```

Comment:
This should have been a query param. Best practice is any filter is implemented via query param. `?olderThan={date}`